src/DynamoCoreWpf/Views/Preview/PreviewControl.xaml.cs

```diff
@@ -499,143 +502,105 @@ private Size ContentToControlSize(Size size)
             return size;
         }
 
-        private void UpdateAnimatorTargetSize(SizeAnimator animator, Size targetSize)
-        {
-            string widthAnimator = string.Empty;
-            string heightAnimator = string.Empty;
-
-            switch (animator)
-            {
-                case SizeAnimator.PhaseIn:
-                    widthAnimator = Element.PhaseInWidthAnimator;
-                    heightAnimator = Element.PhaseInHeightAnimator;
-                    break;
+        #endregion
 
-                case SizeAnimator.Expansion:
-                    widthAnimator = Element.ExpandWidthAnimator;
-                    heightAnimator = Element.ExpandHeightAnimator;
-                    break;
+        #region Private Class Methods - Transition Helpers
 
-                case SizeAnimator.Condensation:
-                    widthAnimator = Element.CondenseWidthAnimator;
-                    heightAnimator = Element.CondenseHeightAnimator;
-                    break;
+        private async void BeginFadeInTransition()
+        {
+            if (!IsHidden) throw new InvalidOperationException();
 
-                case SizeAnimator.Resizing:
-                    widthAnimator = Element.GridWidthAnimator;
-                    heightAnimator = Element.GridHeightAnimator;
-                    break;
+            // do not use delay in tests
+            if (DynamoModel.IsTestMode)
+            {
+                ProcessFadeIn();
+                return;
             }
 
-            sizeAnimators[widthAnimator].To = targetSize.Width;
-            sizeAnimators[heightAnimator].To = targetSize.Height;
-        }
+            // show preview bubble only if mouse stays over node longer than delay time
+            // so that during quick mouse moving preview bubbles won't be shown
+            var delayTimer = new DispatcherTimer(DispatcherPriority.Normal)
+            {
+                Interval = TimeSpan.FromMilliseconds(500)
+            };
 
-        #endregion
+            nodeViewModel.OnMouseLeave += delayTimer.Stop;
```

Comment:
again for these subscriptions, do we need to unsubscribe from them at some point?